### PR TITLE
Update/use connection notices to my jetpack connection wrapper

### PR DESCRIPTION
**Pull request by CodeyGuyDylan**
## Proposed changes:

* Allow `useMyJetpackConnection` to accept additional props that get passed to `useConnection`
* Replace all uses of `useConnection` in My Jetpack to `useMyJetpackConnection`

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
Note: A good way to test these changes is to run it locally and log out the values returned by `useMyJetpackConnection` to make sure they are correct
2. Go to the following pages in My Jetpack to ensure the information looks correct as it relates to the connection

- `/wp-admin/admin.php?page=my-jetpack#/add-license` (might want to have an unattached license for this one)

image

 - `/wp-admin/admin.php?page=my-jetpack#/`

image

 - `/wp-admin/admin.php?page=my-jetpack#/add-jetpack-ai`

image

 - `/wp-admin/admin.php?page=my-jetpack#/jetpack-ai`

image

 - Check out some other interstitials as the tables and cards use the hook
 - `/wp-admin/admin.php?page=my-jetpack#/redeem-token`
image

### CODE REVIEW

**Review by robertsreberski — APPROVED** [review]
We could wait with making that updates until I'll merge `feature/welcome-flow` that enhances the hook:
https://github.com/Automattic/jetpack/blob/42731eb5fd61f3c97c3ebcf372dc3dfba549e9e4/projects/packages/my-jetpack/_inc/hooks/use-my-jetpack-connection/index.ts#L28

As actually from all use cases, we don't have to accept all connection props, but only two: 
```
skipUserConnection = false,
redirectUri = '',
```

And auto-configure the rest.

But in general this PR tests well, so I give it an approval 🟢

**Comment by CodeyGuyDylan** [comment]
> We could wait with making that updates until I'll merge `feature/welcome-flow` that enhances the hook:

Ok ill wait until that is merged and then i'll rebase this 😄

**Pull request closed without merging**
